bin/runtime-common/src/messages.rs

```diff
@@ -288,6 +289,76 @@ pub mod target {
 			}
 		}
 	}
+
+	/// Verify proof of Bridged -> This chain messages.
+	pub fn verify_messages_proof<B: MessageBridge, ThisRuntime>(
+		proof: FromBridgedChainMessagesProof<B>,
+	) -> Result<ProvedMessages<Message<BalanceOf<BridgedChain<B>>>>, &'static str>
+	where
+		ThisRuntime: pallet_substrate_bridge::Trait,
+		ThisRuntime: pallet_message_lane::Trait<MessageLaneInstanceOf<BridgedChain<B>>>,
+		HashOf<BridgedChain<B>>:
+			Into<bp_runtime::HashOf<<ThisRuntime as pallet_substrate_bridge::Trait>::BridgedChain>>,
+	{
+		let (bridged_header_hash, bridged_storage_proof, lane, begin, end) = proof;
```

Comment:
nit: 
```suggestion
		let (bridged_header_hash, bridged_storage_proof, lane_id, begin, end) = proof;
```

Then you can do `let message_key = MessageKey { lane_id, nonce };` later on